src/FieldEncryptProcessEntities.php

```diff
@@ -335,6 +329,58 @@ protected function allowEncryption(ContentEntityInterface $entity, $field_name,
   }
 
   /**
+   * Render a placeholder value to be stored in the unencrypted field storage.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity to encrypt fields on.
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The field to encrypt.
+   * @param string $property_name
+   *   The property to encrypt.
+   *
+   * @return mixed
+   *   The unencrypted placeholder value.
+   */
+  protected function getUnencryptedPlaceholderValue(ContentEntityInterface $entity, FieldItemListInterface $field, $property_name) {
+    $unencrypted_storage_value = NULL;
+
+    $property_definitions = $field->getFieldDefinition()->get('fieldStorage')->getPropertyDefinitions();
+    $data_type = $property_definitions[$property_name]->getDataType();
+
+    switch ($data_type) {
+      case "string":
+      case "email":
+      case "datetime_iso8601":
+      case "duration_iso8601":
+      case "uri":
+      case "filter_format":
+        // Decimal fields are string data type, but get stored as number.
+        if ($field->getFieldDefinition()->getType() == "decimal") {
+          $unencrypted_storage_value = 0;
+        }
+        else {
+          $unencrypted_storage_value = '[ENCRYPTED]';
+        }
+        break;
+
+      case "integer":
+      case "boolean":
+      case "float":
+        $unencrypted_storage_value = 0;
+        break;
+    }
```

Comment:
I'm not sure what kind of default value we would need to provide for the "any" type. As the name suggests, it's not really anything specific,  but just a list of random data, from what I can gather.

The "map" field type will probably be a bit tricky, since we can't really encrypt an array value, unless we serialize it first, which might cause problems, since we're effectively altering the data.

I think I checked entity_reference, and it just results in an integer value (the target ID). Will double check to make sure.